index.html

```diff
@@ -1,126 +1,147 @@
+
+      div:nth-of-type(3) {
+        background: #aaa;
+        border-top: 1px solid #f00;
+      }
+
+      div:nth-of-type(4) {
+        background: #ddd;
+        border-top: 1px solid #f00;
+        border-left: 1px solid #f00;
+      }
+
+      input {
+        width: 100%;
+        font-weight: bold;
+      }
+
+      element-persister {
+        height: 100%;
+      }
+      textarea {
+        width: 100%;
+        height: 80%;
+      }
+      .textwrapper {
+        border: 1px solid #999999;
+        margin: 5px 0;
+        padding: 3px;
+      }
+    </style>
+  </head>
+  <body>
+    <element-persister>
+      <div>
+        <input data-persistance-id="h_ul" />
+        <textarea data-persistance-id="ta_ul"></textarea>
+      </div>
+      <div>
+        <input data-persistance-id="h_ur" />
+        <textarea data-persistance-id="ta_ur"></textarea>
+      </div>
+      <div>
+        <input data-persistance-id="h_bl" />
+        <textarea data-persistance-id="ta_bl"></textarea>
+      </div>
+      <div>
+        <input data-persistance-id="h_br" />
+        <textarea data-persistance-id="ta_br"></textarea>
+      </div>
+    </element-persister>
+  </body>
+
+  <script type="module">
+    const debounce = (wait, callback) => {
+      let timeout = null;
+      return (...args) => {
+        const next = () => callback(...args);
+        clearTimeout(timeout);
+        timeout = setTimeout(next, wait);
+      };
+    };
+
+    class ElementPersister extends HTMLElement {
+      PERSISTANCE_ID_ATTR_NAME = "data-persistance-id";
+
+      _getDomElementsToPersist() {
+        const elementsToPersistQuery = `[${this.PERSISTANCE_ID_ATTR_NAME}]:not([${this.PERSISTANCE_ID_ATTR_NAME}=""])`;
+        const elementsToPersist = Array.from(
+          this.querySelectorAll(elementsToPersistQuery)
+        );
+        return elementsToPersist.filter((element) =>
+          element.hasAttribute(this.PERSISTANCE_ID_ATTR_NAME)
+        );
+      }
+
+      _setPersistedValue(domElement) {
+        const persistanceId = domElement.getAttribute(
+          this.PERSISTANCE_ID_ATTR_NAME
+        );
+        const persistedValue = localStorage.getItem(persistanceId) ?? "";
+        domElement.value = persistedValue;
+      }
+
+      _persistValue(domElement) {
+        const persistanceId = domElement.getAttribute(
+          this.PERSISTANCE_ID_ATTR_NAME
+        );
+        localStorage.setItem(persistanceId, domElement.value);
+      }
+
+      _fillDomElementsWithPersistedValues() {
+        const elementsToPersist = this._getDomElementsToPersist();
+        elementsToPersist.forEach((element) =>
+          this._setPersistedValue(element)
+        );
+      }
+
+      _setupDomElementsPersisting() {
+        const changeHandler = (event) => {
+          this._getDomElementsToPersist().forEach((element) =>
+            this._persistValue(element)
+          );
+        };
+
+        this.addEventListener("input", debounce(250, changeHandler));
```

Comment:
Why 250ms ?

Reply:
🤷‍♂️ no idea really. Had to pick some number 😁 Did not really do much research here